runtime/common/src/parachains.rs

```diff
@@ -550,8 +529,13 @@ fn majority_of(list_len: usize) -> usize {
 	list_len / 2 + list_len % 2
 }
 
-fn localized_payload<H: AsRef<[u8]>>(statement: Statement, parent_hash: H) -> Vec<u8> {
+fn localized_payload<S: AsRef<[u8]>, H: AsRef<[u8]>>(
+	statement: Statement,
+	session_index: S,
+	parent_hash: H,
+) -> Vec<u8> {
 	let mut encoded = statement.encode();
+	encoded.extend(session_index.as_ref());
```

Comment:
```suggestion
	session_index.using_encoded(|s| encoded.extend(s));
```

This is zero allocation and presents a cleaner interface to the caller of the function